diff --git a/packages/ollama/0.11/templates/_helpers.tpl b/packages/ollama/0.11/templates/_helpers.tpl
index 9840ab5..bf677df 100644
--- a/packages/ollama/0.11/templates/_helpers.tpl
+++ b/packages/ollama/0.11/templates/_helpers.tpl
@@ -78,3 +78,35 @@ Models mount path
 {{- define "ollama.modelsMountPath" -}}
 {{- printf "%s/models" ( default "/root/.ollama") }}
 {{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "ollama.imagePullSecrets" -}}
+{{/*
+Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
+but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
+Also, we can not use a single if because lazy evaluation is not an option
+*/}}
+{{- if .Values.global }}
+{{- if .Values.global.imagePullSecrets }}
+imagePullSecrets:
+{{- range .Values.global.imagePullSecrets }}
+  {{- $imagePullSecrets := list }}
+  {{- if kindIs "string" . }}
+    {{- $imagePullSecrets = append $imagePullSecrets (dict "name" .) }}
+  {{- else }}
+    {{- $imagePullSecrets = append $imagePullSecrets . }}
+  {{- end }}
+  {{- toYaml $imagePullSecrets | nindent 2 }}
+{{- end }}
+{{- else if .Values.imagePullSecrets }}
+imagePullSecrets:
+    {{ toYaml .Values.imagePullSecrets }}
+{{- end -}}
+{{- else if .Values.imagePullSecrets }}
+imagePullSecrets:
+    {{ toYaml .Values.imagePullSecrets }}
+{{- end -}}
+{{- end -}}
+
